craft_application/commands/lifecycle.py

```diff
@@ -414,6 +399,17 @@ def _run(
         if shell_after:
             _launch_shell()
 
+    @override
+    def _run(
+        self,
+        parsed_args: argparse.Namespace,
+        step_name: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        if self._use_provider(parsed_args=parsed_args):
+            return super()._run(parsed_args=parsed_args, step_name=step_name)
+        return self._run_real(parsed_args=parsed_args, step_name=step_name)
```

Comment:
I am not understanding every aspects of the PR but I am surprise to see this in a specific `*Command` class and not in the `LifecycleCommand` class.

Reply:
The `pack` command has specific logic it needs to run outside of the provider that occurs separately from the rest of the lifecycle commands. For example, only `pack` actually enables the fetch service.